src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -672,6 +717,7 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 	 */
 	public doChangeLanguageFlavor(uri: string, language: string, provider: string): void {
 		if (this._providers.has(provider)) {
+			this._uriToProvider[uri] = provider;
```

Comment:
Hm... what is this mapping used for? I'm thinking if should be consistent with the actual connection, rather than language flavor. For example, if somebody connects an editor to MySQL, then that URI should be claimed by MySQL connection provider, no matter what language flavor user selects in the editor. What is the scenario where we need to map URI to the language flavor?
In reality this should always match connection anyway (since language flavor selection is not allowed), only for disconnected providers we should be setting this. Maybe keep the assignment in ```sendConnectRequest``` and override it here if user explicitly (somehow) changes it?
One thing I'm not sure about is what behavior do we expect, if user chooses the MySQL language flavor for disconnected editor and then connects it to SQL Server? I think it should reset to ```mssql```, as at that time it's kind of guaranteed they will be SQL Server queries.

Reply:
Mapping is used for both connected as well as disconnected URI. It maps to the provider, so if connected it will get the provider id for the connection and set to that. Whereas if disconnected it will set to the flavor id that yo selected, which is turn is the provider id. When user does sendConnectRequest it will call this doChangeLanguageFlavor and we need to set the map to keep track of the flavor. Whereas this method will also be called when user explicitly sets it. So putting it at a common place. Yes that will be the expected behavior, if it is connected to a server, it will use that flavor.

Reply:
Yeah, that's what I thought - ```sendConnectRequest``` will call ```doChangeLanguageFlavor```, it's just not that obvious. I was thinking maybe we should do it in both places, in case somebody changes it later and doesn't call ```doChangeLanguageFlavor``` when connecting - we should still have that URI (editor) mapped. But I see how that will be a code duplication, so not really sure how much risk is there to keep it in just one place...